File: app/src/main/java/com/example/okta/moviedb_iak/adapter/AdapterRVTrailerMovie.java
package com.example.okta.moviedb_iak.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.okta.moviedb_iak.R;
import com.example.okta.moviedb_iak.model.trailermovie.Result;

import java.util.List;

/**
 * Created by Okta on 04/02/2018.
 */

public class AdapterRVTrailerMovie extends RecyclerView.Adapter<AdapterRVTrailerMovie.CategoryViewHolder> {

    private Context context;
    private List<Result> results;

    public AdapterRVTrailerMovie(Context context) {
        this.context = context;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

    @Override
    public CategoryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_trailer, parent, false);
        return new AdapterRVTrailerMovie.CategoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(CategoryViewHolder holder, int position) {
        String youtube_url = getResults().get(position).getKey();
        String name = getResults().get(position).getName();
        holder.tvTitle.setText(name);
    }

    @Override
    public int getItemCount() {
        return getResults().size();
    }

    class CategoryViewHolder extends RecyclerView.ViewHolder {
        TextView tvTitle;
        ImageView icon;

        public CategoryViewHolder(View itemView) {
            super(itemView);
            tvTitle = itemView.findViewById(R.id.itemTitleTrailer);
            icon = itemView.findViewById(R.id.itemIconTrailer);

        }
    }
}

File: app/src/main/java/com/example/okta/moviedb_iak/adapter/AdapterRVTrailerMovie.kt
package com.example.okta.moviedb_iak.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.example.okta.moviedb_iak.R
import com.example.okta.moviedb_iak.activity.DetailActivity
import com.example.okta.moviedb_iak.adapter.AdapterRVTrailerMovie.CategoryViewHolder
import com.example.okta.moviedb_iak.model.trailermovie.Result

/**
 * Created by Okta on 04/02/2018.
 */

class AdapterRVTrailerMovie(private val context: Context) : RecyclerView.Adapter<CategoryViewHolder>() {
    var results: List<Result>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_trailer, parent, false)
        return CategoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        val youtube_url = results!![position].key
        val name = results!![position].name
        holder.tvTitle.text = name
        holder.itemView.setOnClickListener {
            val id = results!![position].key.toString()
            DetailActivity.watchYoutubeVideo(context, id)
        }
    }

    override fun getItemCount(): Int {
        return results!!.size
    }

    inner class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvTitle: TextView
        var icon: ImageView

        init {
            tvTitle = itemView.findViewById(R.id.itemTitleTrailer)
            icon = itemView.findViewById(R.id.itemIconTrailer)

        }
    }
}
